packages/big-design/src/components/Table/types.ts

```diff
@@ -21,6 +21,7 @@ export interface TableSortable<T> {
 export interface TableItem {
   id?: string | number;
   [key: string]: any;
+  expandableContent?: ReactNode;
```

Comment:
We shouldn't be adding additional properties to the `TableItem` prop. Instead we should think about making the table expandable from the `TableColumn` perspective. I don't see the need of individually adding it per item.

Think about something like this:
```tsx
const columns = [
  { header: 'ID', hash: 'id', expandedContent: ({ id }) => <ComponentThatFetchesById id={id} /> },
]
```